### PR TITLE
Translatable Reference Data

### PR DESCRIPTION
**Pull request by lshepstone**
Stab at implementing a mixin for reference data models to support translation by way of introducing a `cy_name` attribute. Currently, this adds translatable support to the other advice method model, in anticipation of it's use for #5725.

### CODE REVIEW

File: app/models/translatable.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module Translatable
+  def en_name
```
Comment (benlovell) [inline app/models/translatable.rb]: Can we replace this with `alias :en_name :name`

File: app/models/translatable.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    self.name
+  end
+
+  def localized_name
```
Comment (benlovell) [inline app/models/translatable.rb]: Indent the `when` at the same level as `case` and instead of `||`ing, have `name` as the `else` within the `case`. E.g.

```ruby
case I18n.locale
when :en
  en_name
when :cy
  cy_name
else
  name
end
```

Also, it may be more defensive to explicitly bomb out if you hit the `else` clause. These kind of silent, soft-failures can cause surprises.

File: spec/dummy/config/locales/cy.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+cy:
```
Comment (benlovell) [inline spec/dummy/config/locales/cy.yml]: Is this file necessary?

File: spec/factories/other_advice_method.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 FactoryGirl.define do
   factory :other_advice_method do
     name { Faker::Lorem.sentence }
+    sequence(:cy_name) { |n| "Dull gyngor arall #{n}" }
```
Comment (benlovell) [inline spec/factories/other_advice_method.rb]: What's the rationale for making this a sequence?

File: spec/dummy/config/locales/cy.yml (hunk 3 above)
Comment (lshepstone) [inline spec/dummy/config/locales/cy.yml]: Yeah, otherwise when specs run it complains about there being no "cy" locale. Any ideas around that?

Comment (benlovell) [inline spec/dummy/config/locales/cy.yml]: There is, but I guess this is the easiest approach in this case. Ignore that comment :smile:

File: spec/factories/other_advice_method.rb (hunk 4 above)
Comment (lshepstone) [inline spec/factories/other_advice_method.rb]: Realistically reference data entries should be unique, so this ensures they are - same as you did for the accreditation and professional body models etc.

Comment (benlovell) [inline spec/factories/other_advice_method.rb]: I added the sequence so I could uniquely identify the options by name in the specs. Also, the `name` here isn't a sequence so it is somewhat contradictory.

File: app/models/translatable.rb (hunk 2 above)
Comment (lshepstone) [inline app/models/translatable.rb]: So the intention here was not quite to default to name for an unrecognised locale, but rather to ensure that a value is returned if none was found. I agree though, that behaviour probably is unexpected, so can be removed.

File: spec/factories/other_advice_method.rb (hunk 4 above)
Comment (lshepstone) [inline spec/factories/other_advice_method.rb]: Just not sure how to generate unique Welsh sentences :smile:

Comment (benlovell) [inline spec/factories/other_advice_method.rb]: :smile: My point is, if we don't care about `name` being unique, we shouldn't be caring about `cy_name`'s uniqueness or lack thereof. The reason I'm being pushy with this is that this kind of thing tends to get cargo-culted around the rest of the project, and we'll end up with sequences where they're not being used for any specific reason.

Comment (lshepstone) [inline spec/factories/other_advice_method.rb]: Sure, no problem - was looking at Faker's support for Welsh. None. I'll revert to the same as `name`. :smile:

**Comment by benlovell** [comment]
:ship: :shipit:
